Modules/mmapmodule.c

```
@@ -1117,6 +1117,46 @@ mmap_mmap_seek_impl(mmap_object *self, Py_ssize_t dist, int how)
     return NULL;
 }
 
+/*clinic*/
+
+/*[clinic input]
+mmap.mmap.set_name
+
+    name: str
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+mmap_mmap_set_name_impl(mmap_object *self, const char *name)
+/*[clinic end generated code: output=1edaf4fd51277760 input=6c7dd91cad205f07]*/
+{
+    const char* prefix = "cpython:mmap:";
+    if (strlen(name) + strlen(prefix) > 80) {
+        PyErr_SetString(PyExc_ValueError, "name is too long");
+        return NULL;
+    }
+#ifdef MAP_ANONYMOUS
+    if (self->flags & MAP_ANONYMOUS) {
+        char buf[80] = {0, };
+        sprintf(buf, "%s%s", prefix, name);
+        _PyAnnotateMemoryMap(self->data, self->size, buf);
+    }
+    else {
+        /* cannot name non-anonymous mappings */
+        PyErr_SetString(PyExc_ValueError,
+                        "Cannot set annotation on non-anonymous mappings");
+        return NULL;
+    }
+#else
+    /* naming not supported on this platform */
+    PyErr_SetString(PyExc_ValueError,
```

In ssl.c we usually raise a NotImplementedError in such cases. If you always want to raise the same error, I suggest that you document this ValueError.

---

The code was modified to raise NotImplementedError instead.